### ISSUE TITLE
REQ: PLEASE vote for regionalization/translation of Movary

### ISSUE THREAD

**Issue by punkyard** [leepeuker/movary#577]
### Description

hi
yes indeed, I do defend the idea that language is really really important.

Maybe the first reason would be to get film titles in their original version!
Is it because the app is calling theMDB in English that we're getting all film titles in English?

I promise I can tell you -as a member of the french community- SO many people won't use the app I've installed because: they don't read English, they don't like English, they will see no point in adding French movies with american titles to their watch list ..
I'm not commenting that. I'm saying as a fact.

You would get my full support -as a teacher in public schools- in the task of translating your app to French ✌️

a pull request, could do?

### Desired Behavior

choose language when creating user account
choose language when inviting user
get original titles and details in the user language

**Comment by leepeuker** [comment]
Movary in its current state is mainly targeting people who will run an instance for themself. I think most of this target group is fine with having only english (mainly because of personal experience and that no one requested the feature yet).
But I agree that translations are important down the road, to appeal to a wider audience and to enable the existing users to more customization of there personal instance.

I think we can split this topic in three parts:
- language of UI elements
- language of movie meta data
- language of movie titles

Language of UI elements:
We have to through the html templates and javascript code, find all UI texts and create some kind of dictionary. Than we need to setup an easy way for people to add translations to the dictionary.
This should be easy to do on a per user base (=> multiple users using different languages on the same server).

Language of movie (and person?) meta data:
Currently we request and store all data from tmdb in english. If a user changes the language for the movie meta data, we would have to request all the meta data again for the new language (which could take a while for big libraries). If we want to support multiple languages at once, we would have to store and update the meta data for one movie multiple times in each wanted language. So supporting multiple languages for the meta data at the same time is probably complicated and expensive :thinking: 

Languages of movie titles:
I think we already store the original title of movies. It should probably be easy to enable users to display the original title vs the english title. We could add this as a global option and a individual per movie option.

**Summary**
I would recommend to add language options on a per server level and not on a per user level. That is especially because of the issues with the language options for metadata of movies/persons.

**Comment by punkyard** [comment]
all right 💪
I won't be much of a help in the dev, coz unfortunately I can't do that
but in the french translation ^^ and ideas ^^

keep me posted ✌️

**Comment by punkyard** [comment]
> I would recommend to add language options on a per server level and not on a per user level. That is especially because of the issues with the language options for metadata of movies/persons.

When I think of how much people travel these days, and how people connect to each other and exchange - I'd vote for a per user, after thinking

**Comment by leepeuker** [comment]
> When I think of how much people travel these days, and how people connect to each other and exchange - I'd vote for a per user, after thinking

True, but because of the reasons I listed this is sadly relatively complicated and expensive to achieve. The core philosophy of Movary is to have all data locally, this conflicts with dynamic changes of multiple languages.

**Comment by punkyard** [comment]
ok I understand - respecting infrastructure and stability is prioritary ✌️

**Comment by punkyard** [comment]
hi
to keep an eye on this topic, do i have to come back here or check some other page?

**Comment by leepeuker** [comment]
Any updates about this topic will be collected/mentioned here, so it is enough to follow this issue
